### ISSUE TITLE
Installation instructions

### ISSUE THREAD

**Issue by cipher1024** [dlfcn-win32/dlfcn-win32#53]
What is the recommended way of installing this library into a project?

**Comment by traversaro** [comment]
I think the "recommended way" really depends on which build  system you use in your project, and how you handle other 3rd-party dependencies. 

If you use CMake, you can install `dlfcn-win32` in a specific prefix using `CMAKE_INSTALL_PREFIX` , and then find it  using the instructions documented in https://github.com/dlfcn-win32/dlfcn-win32#using-cmake . 
I think it should be possible to also consume it via the  `FetchContent` module, but that is still a work in progress ( https://github.com/dlfcn-win32/dlfcn-win32#using-cmake ).

If you tipically vendor  other dependencies, you can just grab the `dlfcn.c` and `dlfcn.h` files and include them in your project. To comply to the LGPL license of the project  and avoid the need to license under the LGPL all your project, I suggest to make sure that you compile the `dlfcn.c` in a shared library.

**Comment by cipher1024** [comment]
So far so good. I followed the instruction for `cmake` and I added a call to `vcpkg` in my appveyor script. `cmake` doesn't seem to be finding `dlfcn-win32`. How do I encode the path to `dlfcn-win32` into my `cmake` file?

**Comment by traversaro** [comment]
If you included the library via `FetchContent`, the `dlfcn-win32::dl` target should be available after you call `FetchContent_MakeAvailable(dlfcn-win32)`. If this is not working, there is a bug on dlfcn-win32 side. Furthermore, include directory should be already handled by the target, so you should not need `target_include_directories(r-type-utils INTERFACE ${CMAKE_BINARY_DIR}/_deps/dlfcn-win32-src/src)`. Can you remove `find_package` and `target_include_directories` and try again?

**Comment by traversaro** [comment]
> So far so good. I followed the instruction for `cmake` and I added a call to `vcpkg` in my appveyor script. `cmake` doesn't seem to be finding `dlfcn-win32`. How do I encode the path to `dlfcn-win32` into my `cmake` file?

Sorry, this is a really old question, but I missed it. If you installed a library via vcpkg, to find it you need to add `find_package(dlfcn-win32)` in your CMakeLists.txt, and call your cmake by passing the vcpkg toolchain, see https://learn.microsoft.com/en-us/vcpkg/users/buildsystems/cmake-integration#cmake_toolchain_file for more details.

**Comment by traversaro** [comment]
You are right, there is something missing on our side (that issue is actually tracked in https://github.com/dlfcn-win32/dlfcn-win32/issues/49). As a temporary workaround, you can just change `    target_link_libraries(r-type-utils INTERFACE dlfcn-win32::dl)` to `    target_link_libraries(r-type-utils INTERFACE dl)` and it should work fine (I guess).
